### ISSUE TITLE
Audio Sprites demo not working in recent MacOS Safari and iOS updates

### ISSUE THREAD

**Issue by sfxxx** [goldfire/howler.js#1535]
To my dismay, I learned that (Howler based) audio playback stopped working for some users of my website (http://symres.org), after updating their iPhones or desktop Safari browsers. Since I don't have an iPhone, I need to rely or user reports, but I can reproduce it in macOS Safari - no errors or JS console messages, just no playback.

My site's dysfunction seems to correlate with that of Howler's own Audio Sprites demo. So I hope it can save myself the effort to extract demo code for reproducing the problem (though it's just straightforward playback of loops, either immediately or in the load event handler; but I do assign to h._sprite.__default in order to produce smooth loop transitions, which might be something the Audio Sprites demo might do, too) 

I reproduced the Sprites Demo problem in Safari 15.1 on macOS 10.15.7. According to my user, it also fails on his iPhone under both iOS 14.8 and 15.1 in Safari as well as Chrome(!). Strangely enough it _does_ work on my iPad in Safari on iPadOS 14.71 and 15.1.

Googling for similar issues brought up #1407. I tried the suggestion to use html5: true, which indeed makes playback work again in my desktop Safari. However, loops then no longer play seamlessly, nor does my loop synchronization code work, so it's not a solution.

**Comment by evolross** [comment]
I'm noticing that non-sprite sounds are not working Mac OS Safari 15. Have to add `html: true` to get them to work.

**Comment by evolross** [comment]
I thought I would add, my quick and dirty way of detecting Mac OS Safari looks like this:
```
if(navigator.userAgent.indexOf("Safari") != -1 && navigator.userAgent.indexOf("Chrome") === -1 && navigator.userAgent.indexOf("iPhone") === -1)
   useHTML5 = true;
```
It's weird because the `userAgent` for Chrome contains the word "Safari", but the `userAgent` for Safari only contains "Safari".

**Comment by sfxxx** [comment]
Update to the OP: Sprites demo also doesn't work on Chrome under iOS 15.1.

**Comment by lustremedia** [comment]
All Browsers on iOS are basically "Safari" it does not matter if they are Chrome or Firefox, underneath is a Safari.

https://adactio.medium.com/web-browsers-on-ios-b120a9a75cce

**Comment by sfxxx** [comment]
Thanks for teaching me something.

**Comment by HorusAK** [comment]
Safari 15 with macOS Catalina 10.15 have a regression that doesn't decode audio.

**Comment by sfxxx** [comment]
Thanks. Do you mean decoding of compressed audio? So that using WAV might be a temporary workaround? Do you have a source for your statement?

**Comment by goldfire** [comment]
It looks like this should be fixed in the next Safari update. Here is the webkit issue that is already marked as resolved in nightly builds: https://bugs.webkit.org/show_bug.cgi?id=230974.

**Comment by sfxxx** [comment]
Not fixed in recently released MacOs Safari 15.2.

**Comment by goldfire** [comment]
> Not fixed in recently released MacOs Safari 15.2.

I've been doing further digging on this and just can't see any way that howler can address this other than by forcing HTML5 audio on that specific browser/OS combination, though that can easily be done on the user side by passing true/false to the `html5` param based on browser/OS. The issue happens within `decodeAudioData`, so it is on Apple to fix.
